internal/databases/postgres/tarball_streamer.go

```diff
@@ -91,24 +91,22 @@ func NewTarballStreamer(input io.Reader, maxTarSize int64, bundleFiles BundleFil
 	return streamer
 }
 
-//NextFile is what makes the TarballStreamer move to the next file.
+//NextInputFile is what makes the TarballStreamer move to the next file.
 func (streamer *TarballStreamer) NextInputFile() (err error) {
 	// First output tar, or switching to next
 	if streamer.outputTar == nil {
 		streamer.outputTar = tar.NewWriter(streamer.outputIo)
 		streamer.tarFileReadIndex = 0
 		if streamer.curHeader != nil {
-			err = streamer.outputTar.WriteHeader(streamer.curHeader)
-			if err != nil {
-				return err
-			}
+			return streamer.addFile()
```

Comment:
Can you expand on why this is necessary? I see `addFile` starts with WriteHeader, so what was causing the rest to sometimes be necessary but usually safe to skip?

Reply:
We should always use addFile, because that also adds to the Tee file if needed and all.
But the bigger issue why this was breaking was there was no return, meaning we continued in this method, which does a streamer.inputTar.Next() skipping the body of the current file, and moving straight to the next file...